### ISSUE TITLE
IS-04-01 test_16_01 timeout

### ISSUE THREAD

**Issue by garethsb-sony** [AMWA-TV/nmos-testing#312]
We wonder if we ought to increase the acceptable timeout on connection failure to match the garbage collection interval (12s) rather than the heartbeat interval (5s), since any connection timeout up to the larger value seems like a possible strategy for Nodes.

There is already a longer timeout here when `ENABLE_HTTPS = True` due to empirical testing with the test suite in some environments, which needs to be revisited.

**Comment by prince-chrismc** [comment]
I would like to put forth an argument for increasing the timeout delay. The default settings for both Windows and Linux are greater then the test allows and produce a warning.

For windows the default connection timeout is 21s ( initialRto is 3s and the SYN re-transmission is 3 as well ) The TCP RFC specified minimum for SYN re-transmission is 3.

 For Linux the connection timeout accord to the [man page](http://man7.org/linux/man-pages/man7/tcp.7.html) is 
> tcp_syn_retries [...] The default value is 5, which corresponds to approximately 180 seconds.

**Comment by garethsb-sony** [comment]
That's true, but given the NMOS spec for garbage collection, I think the Node is supposed to time out the connection attempt earlier than that, to ensure that it has a chance to try the next Registry within the GC interval.

**Comment by andrewbonney** [comment]
Yes, much of the reason this test was added was to ensure that default or potentially infinite timeouts weren't in use for this specific interaction.

**Comment by prince-chrismc** [comment]
Perhaps this behavior could be mentioned in the [IS-04 docs 4.1 registration behavior ](https://github.com/AMWA-TV/nmos-discovery-registration/blob/v1.2.x/docs/4.1.%20Behaviour%20-%20Registration.md)?
